pallets/lbp/src/lib.rs

```diff
@@ -0,0 +1,987 @@
+		SaleStarted,
+		/// Sale is still in progress
+		SaleNotEnded,
+		/// Sale is not running
+		SaleIsNotRunning,
+		/// Sale already ended
+		CannotPauseEndedPool,
+		/// Sale already ended
+		CannotUnpauseEndedPool,
+		/// Sale is already paused
+		CannotPausePausedPool,
+		/// Pool cannot be paused
+		PoolIsNotPausable,
+		/// Pool is not paused
+		PoolIsNotPaused,
+		/// Sale duration is too long
+		MaxSaleDurationExceeded,
+		/// Liquidity being added should not be zero
+		CannotAddZeroLiquidity,
+		/// Can not remove zero liquidity
+		CannotRemoveZeroLiquidity,
+		/// Overflow after adding liquidity
+		LiquidityOverflow,
+		/// Asset balance too low
+		InsufficientAssetBalance,
+		/// Pool does not exist
+		PoolNotFound,
+		/// Pool has been already created
+		PoolAlreadyExists,
+		/// Pool does not contain the asset
+		InvalidAsset,
+		/// Invalid block number
+		InvalidBlockNumber,
+		/// Calculation error
+		WeightCalculationError,
+		/// Invalid block number
+		BlockNumberInvalid,
+		/// Can not perform a trade with zero amount
+		ZeroAmount,
+		/// Trade amount is too high
+		MaxInRatioExceeded,
+		/// Trade amount is too high
+		MaxOutRatioExceeded,
+		/// Invalid fee amount
+		FeeAmountInvalid,
+		/// Trading limit reached
+		AssetBalanceLimitExceeded,
+		/// An unexpected integer overflow occurred
+		Overflow,
+		/// Nothing to update
+		NothingToUpdate,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Pool was created by the `CreatePool` origin. [who, pool_id, asset_a, asset_b, amount_a, amount_b]
+		PoolCreated(T::AccountId, PoolId<T>, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Pool data were updated. [who, pool_id]
+		PoolUpdated(T::AccountId, PoolId<T>),
+
+		/// New liquidity was provided to the pool. [who, asset_a, asset_b, amount_a, amount_b]
+		LiquidityAdded(T::AccountId, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Liquidity was removed from the pool. [who, asset_a, asset_b, amount_a, amount_b]
+		LiquidityRemoved(T::AccountId, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Pool was destroyed. [who, asset_a, asset_b, balance_a, balance_b]
+		PoolDestroyed(T::AccountId, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Sale executed. [who, asset_in, asset_out, amount, sale_price]
+		SellExecuted(T::AccountId, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Purchase executed. [who, asset_out, asset_in, amount, buy_price]
+		BuyExecuted(T::AccountId, AssetId, AssetId, BalanceOf<T>, BalanceOf<T>),
+
+		/// Pool was paused. [who, pool_id]
+		Paused(T::AccountId, PoolId<T>),
+
+		/// Pool was unpaused. [who, pool_id]
+		Unpaused(T::AccountId, PoolId<T>),
+	}
+
+	/// Paid deposit. Returned when a pool is destroyed and removed from the storage.
+	#[pallet::storage]
+	#[pallet::getter(fn pool_deposit)]
+	pub type PoolDeposit<T: Config> = StorageMap<_, Blake2_128Concat, PoolId<T>, BalanceOf<T>, ValueQuery>;
```

Comment:
If nothing else is going to be done with this PoolDeposit - it is actually useless. Because currently it only stores same amount for every pool ( pallet's config option), and it is only removed from storage.

So you can simply remove it completely. Because you wont' allow to create a pool if `T::MultiCurrency::reserve` fails.

Reply:
Regarding the storage, (afaik) this pattern is commonly used in substrate pallets. This way it's possible to change the deposit value in the future, keep track of previous deposits, and charge the difference if necessary.
But in the case of LBP, the deposit functionality has been removed.